source/common/router/router.cc

```diff
@@ -1038,7 +1038,7 @@ void Filter::onStreamMaxDurationReached(UpstreamRequest& upstream_request) {
       StreamInfo::ResponseFlag::UpstreamMaxStreamDurationReached);
   // sendLocalReply may instead reset the stream if downstream_response_started_ is true.
   callbacks_->sendLocalReply(
-      Http::Code::RequestTimeout, "upstream max stream duration reached", modify_headers_,
+      Http::Code::ServiceUnavailable, "upstream max stream duration reached", modify_headers_,
```

Comment:
Should we return 504 instead of 503 here similar to what we do for onResponseTimeout()? Or this case little different?

Reply:
Yeah, I think that's a better choice. Updating